packages/preact-iso/src/client/components.tsx

```diff
@@ -0,0 +1,50 @@
+import { h } from 'preact'
+import { forwardRef } from 'preact/compat'
+import { useLocation } from 'preact-iso'
+import { LinkProps, To } from './types'
+
+type LinkRef = import('preact').RefObject<HTMLAnchorElement>
+
+export const components = <Path extends string, Params extends Record<string, any>>() => {
+  return {
+    // @ts-expect-error - Type complexity with forwardRef and generics
+    Link: forwardRef(({ to, params, ...props }: LinkProps<Path | To<Path>, Params>, ref: any) => {
```

Comment:
You may need to add a click handler on links to stop event propagation; iso attaches a delegated handler to the window to capture any bubbling link clicks and it's likely the route patterns it uses don't align with the patterns used here (most libs do things a bit differently).